specification/elasticsan/resource-manager/Microsoft.ElasticSan/stable/2023-01-01/elasticsan.json

```diff
@@ -1960,6 +2312,119 @@
         "id"
       ],
       "description": "Virtual Network rule."
+    },
+    "Snapshot": {
+      "type": "object",
+      "description": "Response for Volume Snapshot request.",
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Properties of Volume Snapshot.",
+          "$ref": "#/definitions/SnapshotProperties"
+        },
+        "systemData": {
+          "description": "Resource metadata required by ARM RPC",
+          "readOnly": true,
+          "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/systemData"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        }
+      ]
+    },
+    "SnapshotCreateParameter": {
```

Comment:
>SnapshotCreateParameter [](http://example.com/codeflow?start=5&length=23)

How is this different from definitions/Snapshot? Recommend unifying on Snapshot: they should be the same anyway.

Reply:
They both are different, SnapshotCreateParameter Model has required = CreationData while Snapshot Model has nothing as required.
The idea to have a separate model for Create operation was to take care of required properties.

Reply:
Would it be incrementally better to inherit allOf Snapshot, but just add some requiredness onto that?